component/src/main/java/org/wso2/extension/siddhi/execution/env/ResourceBatchWindowProcessor.java

```diff
@@ -0,0 +1,279 @@
+                if (scheduler != null) {
+                    scheduler.notifyAt(nextEmitTime);
+                }
+            }
+            ComplexEventChunk<StreamEvent> outputStreamEventChunk = new ComplexEventChunk<StreamEvent>(true);
+            while (streamEventChunk.hasNext()) {
+                StreamEvent streamEvent = streamEventChunk.next();
+                StreamEvent clonedStreamEvent = streamEventCloner.copyStreamEvent(streamEvent);
+                int groupEventMapSize = 0;
+                if (!streamEvent.getType().equals(ComplexEvent.Type.TIMER)) {
+                    Object groupEventMapKey = groupKeyExpressionExecutor.execute(clonedStreamEvent);
+                    if (groupEventMap.containsKey(groupEventMapKey)) {
+                        groupEventMap.get(groupEventMapKey).streamEventList.add(clonedStreamEvent);
+                    } else {
+                        List<StreamEvent> list = new ArrayList<>();
+                        list.add(clonedStreamEvent);
+                        groupEventMap.put(groupEventMapKey, new ResourceStreamEventList(list,
+                                clonedStreamEvent.getTimestamp() + timeInMilliSeconds));
+                    }
+                    groupEventMapSize = groupEventMap.get(groupEventMapKey).streamEventList.size();
+                }
+                for (Map.Entry<Object, ResourceStreamEventList> entry : groupEventMap.entrySet()) {
+                    if (groupEventMapSize == windowLength || entry.getValue().expiryTimestamp <
+                            currentTime) {
+                        //update current event chunk with event batch
+                        for (StreamEvent event : entry.getValue().streamEventList) {
+                            currentEventChunk.add(event);
+                        }
+                        groupEventMap.remove(entry.getKey());
+                        //update outputStreamEventChunk with expired events chunk
+                        if (outputExpectsExpiredEvents) {
+                            if (eventsToBeExpired.getFirst() != null) {
+                                while (eventsToBeExpired.hasNext()) {
+                                    StreamEvent expiredEvent = eventsToBeExpired.next();
+                                    expiredEvent.setTimestamp(currentTime);
+                                }
+                                outputStreamEventChunk.add(eventsToBeExpired.getFirst());
+                            }
+                        }
+                        if (eventsToBeExpired != null) {
+                            eventsToBeExpired.clear();
+                        }
+                        //update outputStreamEventChunk and eventsToBeExpired with current event chunk
+                        if (currentEventChunk.getFirst() != null) {
+                            // add reset event in front of current events
+                            outputStreamEventChunk.add(resetEvent);
```

Comment:
get the first expired event of each chunk and add that as the reset event

Reply:
Added fixes: 8eebb1fb5252eee92a81e76203463c40fbf49682